haskell/test/SMT/Common/Run.hs

```diff
@@ -0,0 +1,219 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module SMT.Common.Run
+  ( runBaseDaoPropertyTest
+  ) where
+
+import Universum hiding (drop, swap)
+
+import qualified Data.Map as Map
+import Data.Sequence (Seq(..))
+import Fmt (pretty)
+import Hedgehog
+
+import Lorentz hiding (now, (>>))
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import Lorentz.Test.Consumer (contractConsumer)
+import Michelson.Typed.Convert (convertContract, untypeValue)
+import Michelson.Untyped.Entrypoints (unsafeBuildEpName)
+import Morley.Nettest
+import qualified Morley.Micheline as MM
+import Morley.Nettest.Pure
+
+import Ligo.BaseDAO.Contract
+import Ligo.BaseDAO.Types
+import SMT.Model.BaseDAO.Contract
+import SMT.Model.BaseDAO.Types
+import SMT.Common.Gen
+import SMT.Common.Types
+import Test.Ligo.BaseDAO.Common.StorageHelper as StorageHelper
+import Test.Ligo.BaseDAO.Common
+
+
+runBaseDaoPropertyTest :: MkPropertyTestOption -> PropertyT IO ()
+runBaseDaoPropertyTest option@MkPropertyTestOption{..} = do
+  -- PropertyTestInput (contractCalls, ms) <- forAll (runGeneratorT (genPropertyTestInput proposalMeta) initGeneratorState)
+
+  mkPropertyTestInput <- forAll (runGeneratorT (genPropertyTestInput option) $ initGeneratorState mptoMkPropose)
+  nettestTestProp $
+    (uncapsNettestEmulated $ do
+        -- Set initial level
```

Comment:
I think this refers to the  `advanceLevel` that is called a few lines below?

Reply:
Fixed.